### PR TITLE
Fix critical and high fossa scan issues

### PR DESCRIPTION
**Pull request by ShalkiWenushika**
### Purpose
$Subject

### Issues

Fixes # 

### Automation tests
 - Unit tests added: Yes/No
 - Integration tests added: Yes/No

### Tested environments

Not Tested

---
#### Maintainers: Check before merge
- [x] Assigned 'Type' label
- [x] Assigned the project
- [x] Validated respective github issues
- [x] Assigned milestone to the github issue(s)

### CODE REVIEW

**Review by pubudu538 — DISMISSED** [review]

File: adapter/go.mod [hunk 1]
```diff
@@ -60,7 +60,7 @@ require (
 	github.com/pkg/errors v0.9.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	go.mongodb.org/mongo-driver v1.7.5 // indirect
-	golang.org/x/crypto v0.0.0-20220511200225-c6db032c6c88 // indirect
+	golang.org/x/crypto v0.1.0 // indirect
```
Comment (VirajSalaka) [inline adapter/go.mod:63]: shall we do a go mod tidy and see if there are any go.sum changes to move in.

Comment (ShalkiWenushika) [reply adapter/go.mod:63]: Fixed via https://github.com/wso2/product-microgateway/pull/3303/commits/937f11db1f7192565fbc3ef9ff37db9dbe0c7d36

**Review by pubudu538 — APPROVED** [review]

**Review by VirajSalaka — APPROVED** [review]

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230322.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CentralIN) cluster : stage-deployment-v2 : 20230322.1

**Comment by choreo-cicd** [comment]
[failed] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230322.1

**Comment by choreo-cicd** [comment]
[failed] Controlplane cluster : stage-deployment-v2 : 20230322.2

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(NorthEU) cluster : stage-deployment-v2 : 20230322.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(CentralIN) cluster : stage-deployment-v2 : 20230322.4

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : stage-deployment-v2 : 20230322.4

**Comment by choreo-cicd** [comment]
[succeeded] Dataplane(EastUS) cluster : stage-deployment-v2 : 20230322.4

**Comment by choreo-cicd** [comment]
[failed] Dataplane(NorthEU) cluster : prod-deployment-v2 : 20230327.5

**Comment by choreo-cicd** [comment]
[failed] Dataplane(CentralIN) cluster : prod-deployment-v2 : 20230327.5

**Comment by choreo-cicd** [comment]
[succeeded] Controlplane cluster : prod-deployment-v2 : 20230327.7

**Pull request merged**